plugins/secom_cl.py

```diff
@@ -14,54 +15,69 @@
 ./secom_cl --xrep 45 --yrep 5 --prefix filename-prefix
 
 --prefix indicates the beginning of the filename.
-The files are saved in TIFF, with the y, x positions (in nm) in the name.
+The files are saved in TIFF, with the y, x positions of the ebeam (in nm) in the name,
+the total number of ebeam positions in x and y, the physical distance between positions in x and y
+and the type of the acquisition.
 
-'''
+"""
 
 from __future__ import division
 
-import argparse
 from collections import OrderedDict
+from concurrent.futures._base import CancelledError
 import copy
-import itertools
 import logging
 import math
 import numpy
+
+from odemis.util.filename import guess_pattern, create_filename, update_counter
+
+from odemis.gui.util import call_in_wx_main
+
+from odemis.gui.comp.overlay.world import RepetitionSelectOverlay
+
+from odemis.gui.model import TOOL_ROA, TOOL_RO_ANCHOR, TOOL_NONE
+
 from odemis import dataio, model, util, gui
-from odemis.acq import stream
+from odemis import acq
+from odemis.acq import stream, leech
 from odemis.gui.conf import get_acqui_conf
 from odemis.gui.plugin import Plugin, AcquisitionDialog
 from odemis.util import img
 import os.path
-import sys
-import threading
-import time
+
+import odemis.acq.stream as acqstream
 
 
 # Exposure time of the AR CCD
-EXP_TIME = 1 # s
+EXP_TIME = 1  # s
 # Binning for the AR CCD
-BINNING = (1, 1) # px, px
+BINNING = (1, 1)  # px, px
 
 # file format
 FMT = "TIFF"
 # Filename format
-FN_FMT = u"%(prefix)s_grid=%(xres)dx%(yres)d_stepsize=%(stepsize).2fnm_n=%(idx)03d.tiff"
+FN_FMT = u"%(prefix)s_grid=%(xres)dx%(yres)d_stepsize=%(xstepsize)dx%(ystepsize)dnm_n=%(xpos)dx%(ypos)d_%(type)s.tiff"
+
 
 def get_ccd_md(ccd):
     """
-    Returns the Metadata associated with the ccd, including fine alignment corrections.
+    Returns the Metadata associated with the optical detector, including the fine alignment corrections.
+    :param ccd: (model.Component????) The optical detector.
     """
     # The only way to get the right info is to look at what metadata the
     # images will get
     md = copy.copy(ccd.getMetadata())
-    img.mergeMetadata(md) # apply correction info from fine alignment
+    img.mergeMetadata(md)  # apply correction info from fine alignment
 
     return md
 
+
 def get_ccd_pxs(ccd):
     """
-    Returns the (theoretical) pixelsize of the CCD (projected on the sample).
+    Calculates the (theoretical) pixel size of the optical detector (projected on the sample).
+    :param ccd: (model.Component????) The optical detector.
+    :returns: () The theoretical pixel size of the optical detector.  TODO @Eric isn't that the effective px size?
```

Comment:
I think that "theoretical" here meant that PIXEL_SIZE_COR is not taken into account. You could drop the word, and make the sentence less arguable.